packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -460,18 +460,43 @@ async function fetchDeployments(cloudProviderType: CloudProvider['type'], deploy
       }
 
       const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
+      const latestDepolymentFileNames = Object.keys((stageDirectory.children[latestDeployment] as Directory).children);
+
+      const requiredFileNames = ['config.json', 'secrets.env', 'default.tfstate'];
+      const missingRequiredFiles = requiredFileNames.filter(
+        (requiredFileName) => !latestDepolymentFileNames.includes(requiredFileName)
+      );
+      if (missingRequiredFiles.length) {
+        logger.warn(
+          `Airnode '${airnodeAddress}' with stage '${stage}' is missing files: ${missingRequiredFiles.join(
+            ', '
+          )}. Deployer commands may fail and manual removal may be necessary.`
+        );
+      }
+
```

Comment:
This will detect any of the three missing files, but note that if the missing file is either `secrets.env` or `config.json`, it is not possible to derive the deploymentId so the commands will fail with an error:

`No deployment with ID 'aws...' found`

![image](https://user-images.githubusercontent.com/65347007/208860312-95d69024-2870-4485-a72c-dc797a8cd869.png)